### ISSUE TITLE
beartype 0.10.0

### ISSUE THREAD

**Issue by KyleKing** [beartype/bearboto3#3]
Hi! I love the beartype-ecosystem and want to start using your package, but I'm getting errors that it doesn't support 0.10.0, which I'm using for the new beartype.typing:

Could you relax this version constraint?

https://github.com/beartype/bearboto3/blob/998ee1807792580862620f406f61bff654b7156d/pyproject.toml#L12

```sh
> poetry add bearboto3                                                                                                                     
Using version ^0.1.1 for bearboto3

Updating dependencies

Resolving dependencies... (4.0s)
  SolverProblemError

  Because bearboto3 (0.1.1) depends on beartype (>=0.9.0,<0.10.0)
   and no versions of bearboto3 match >0.1.1,<0.2.0, bearboto3 (>=0.1.1,<0.2.0) requires beartype (>=0.9.0,<0.10.0).
  So, because calcipy depends on both beartype (>=0.10.0) and bearboto3 (^0.1.1), version solving failed.
```

Thanks for building this!!

**Comment by leycec** [comment]
**Sorta my fault.** I'm hanging my head. `beartype` currently follows feels-based versioning rather than [semver](https://semver.org), which is why the `^` operator unexpectedly failed here.

Semver's a great <sup>arguably, mandatory</sup> fit for webdev, where strict version provisioning is the difference between your investors throwing you over a bridge in a sack versus them not doing that. By the same token, semver's *not* so great <sup>arguably, awful</sup> outside of webdev, where strict version provisioning continually breaks compatibility with everything outside a walled garden.

So. `beartype` is *probably* going to continue following feels-based versioning. Explosive tectonic activity ensues. :volcano:

**Comment by studioph** [comment]
Hey thanks for reporting this (and sorry that you ran into trouble!). 

@leycec I'm a little confused on what in `beartype`s "feels-based" versioning (as you call it) breaks the `^` in `bearboto3`s `pyproject.toml` in this circumstance. I'm thinking changing `^` to `>=` here is the proper way to ensure compatibility moving forward?

**Comment by studioph** [comment]
Ah nevermind, I found [this](https://nodesource.com/blog/semver-a-primer/#0xxversions) explanation that `^` does not work with `0.x.x` versions in semver. Learn something new everyday.

**Comment by leycec** [comment]
*...heh.*

**Yup.** `0.x.x` versioning is super-awkward in semver. Unsure why they went there, but they did.

That's why `beartype` currently ignores semver for these early `beartype 0.x.x` release cycles. Once `beartype 1.0.0` lands, we intend to embrace strict semver. Specifically, I do solemnly swear on my step-grandfather's loamy garden ashes never to break backward compatibility **except at major version bumps** (e.g., `beartype 1.0.0`, `beartype 2.0.0`). Major version bumps of `beartype` will *probably* clean house by removing deprecated functionality everyone should have been warned about well in advance. That's pretty much semver if you squint at it, except we ignore semver for these early `beartype 0.x.x` release cycles. Whatevah!

If anyone would ever like us to embrace strict semver before `beartype < 1.0.0`, just drop us a line at our issue tracker. `beartype` *is* increasingly being adopted for high-profile webdev work (e.g., [Polysynth](https://polysynth.com)). So – maybe we want to go there? :man_shrugging:

**Comment by studioph** [comment]
@KyleKing I've just released `0.1.2` which should fix the version constraint issues. I can reopen this if the problem persists.

**Comment by KyleKing** [comment]
Thanks! That seems to work for me!
